File: yandex_backend/52. Камни и украшения/solution.cpp
#include <iostream>
#include <string>
#include <set>

using namespace std;

int main() {
  string j, s;
  getline(cin, j);
  getline(cin, s);

  set<char> jewelry_set(j.begin(), j.end());
  int count = 0;

  for (char c : s) {
    if (jewelry_set.count(c)) {
      count++;
    }
  }

  cout << count << endl;
  return 0;
}

File: yandex_backend/52. Камни и украшения/solution.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution
{
    public static void Main(string[] args)
    {
        string j = Console.ReadLine();
        string s = Console.ReadLine();

        HashSet<char> jewelrySet = new HashSet<char>(j);
        int count = 0;

        foreach (char c in s)
        {
            if (jewelrySet.Contains(c))
                count++;
        }

        Console.WriteLine(count);
    }
}